native/dtrevc3.go

```diff
@@ -0,0 +1,866 @@
+						bi.Dscal(n-ki, rec, b[ki*ldb+iv+1:], ldb)
+						vmax = 1
+						vcrit = bignum
+					}
+					b[j*ldb+iv] -= bi.Ddot(j-ki-2, t[(ki+2)*ldt+j:], ldt, b[(ki+2)*ldb+iv:], ldb)
+					b[j*ldb+iv+1] -= bi.Ddot(j-ki-2, t[(ki+2)*ldt+j:], ldt, b[(ki+2)*ldb+iv+1:], ldb)
+					b[(j+1)*ldb+iv] -= bi.Ddot(j-ki-2, t[(ki+2)*ldt+j+1:], ldt, b[(ki+2)*ldb+iv:], ldb)
+					b[(j+1)*ldb+iv+1] -= bi.Ddot(j-ki-2, t[(ki+2)*ldt+j+1:], ldt, b[(ki+2)*ldb+iv+1:], ldb)
+					// Solve 2×2 complex linear equation
+					//  [ [T[j,j]   T[j,j+1]  ]^T - (wr-i*wi)*I ]*X = scale*b
+					//  [ [T[j+1,j] T[j+1,j+1]]                 ]
+					scale, _, _ := impl.Dlaln2(true, 2, 2, smin, 1, t[j*ldt+j:], ldt,
+						1, 1, b[j*ldb+iv:], ldb, wr, -wi, x[:], 2)
+					// Scale if necessary.
+					if scale != 1 {
+						bi.Dscal(n-ki, scale, b[ki*ldb+iv:], ldb)
+						bi.Dscal(n-ki, scale, b[ki*ldb+iv+1:], ldb)
+					}
+					b[j*ldb+iv] = x[0]
+					b[j*ldb+iv+1] = x[1]
+					b[(j+1)*ldb+iv] = x[2]
+					b[(j+1)*ldb+iv+1] = x[3]
+					vmax01 := math.Max(math.Abs(x[0]), math.Abs(x[1]))
+					vmax23 := math.Max(math.Abs(x[2]), math.Abs(x[3]))
+					vmax = math.Max(vmax, math.Max(vmax01, vmax23))
+					vcrit = bignum / vmax
+					j += 2
+				}
+			}
+			// Copy the vector x or Q*x to VL and normalize.
+			switch {
+			case howmny != lapack.AllEigVecMulQ:
+				// No back-transform: copy x to VL and normalize.
+				bi.Dcopy(n-ki, b[ki*ldb+iv:], ldb, vl[ki*ldvl+is:], ldvl)
+				bi.Dcopy(n-ki, b[ki*ldb+iv+1:], ldb, vl[ki*ldvl+is+1:], ldvl)
+				emax := 0.0
+				for k := ki; k < n; k++ {
+					emax = math.Max(emax, math.Abs(vl[k*ldvl+is])+math.Abs(vl[k*ldvl+is+1]))
+				}
+				remax := 1 / emax
+				bi.Dscal(n-ki, remax, vl[ki*ldvl+is:], ldvl)
+				bi.Dscal(n-ki, remax, vl[ki*ldvl+is+1:], ldvl)
+				for k := 0; k < ki; k++ {
+					vl[k*ldvl+is] = 0
+					vl[k*ldvl+is+1] = 0
+				}
+			case nb == 1:
+				// Version 1: back-transform each vector with GEMV, Q*x.
+				if n-ki-2 > 0 {
+					bi.Dgemv(blas.NoTrans, n, n-ki-2,
+						1, vl[ki+2:], ldvl, b[(ki+2)*ldb+iv:], ldb,
+						b[ki*ldb+iv], vl[ki:], ldvl)
+					bi.Dgemv(blas.NoTrans, n, n-ki-2,
+						1, vl[ki+2:], ldvl, b[(ki+2)*ldb+iv+1:], ldb,
+						b[(ki+1)*ldb+iv+1], vl[ki+1:], ldvl)
+				} else {
+					bi.Dscal(n, b[ki*ldb+iv], vl[ki:], ldvl)
+					bi.Dscal(n, b[(ki+1)*ldb+iv+1], vl[ki+1:], ldvl)
+				}
+				emax := 0.0
+				for k := 0; k < n; k++ {
+					emax = math.Max(emax, math.Abs(vl[k*ldvl+ki])+math.Abs(vl[k*ldvl+ki+1]))
+				}
+				remax := 1 / emax
+				bi.Dscal(n, remax, vl[ki:], ldvl)
+				bi.Dscal(n, remax, vl[ki+1:], ldvl)
+			default:
+				// Version 2: back-transform block of vectors with GEMM.
+				// Zero out above vector.
+				// Could go from ki-nv+1 to ki-1.
+				for k := 0; k < ki; k++ {
+					b[k*ldb+iv] = 0
+					b[k*ldb+iv+1] = 0
+				}
+				iscomplex[iv] = ip
+				iscomplex[iv+1] = -ip
+				iv++
+				// Back-transform and normalization is done below.
+			}
+		}
+		if nb > 1 {
```

Comment:
From L190-195. `nb` is greater than one only if the backtransform is being done and there is enough workspace, otherwise the `case nb == 1` above are taken.